src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -762,25 +802,23 @@ public AbstractStringBuilder append(boolean b) {
         ensureCapacityInternal(count + (b ? 4 : 5));
         int count = this.count;
         byte[] val = this.value;
-        if (isLatin1()) {
-            if (b) {
-                val[count++] = 't';
-                val[count++] = 'r';
-                val[count++] = 'u';
-                val[count++] = 'e';
+        boolean latin1 = isLatin1();
+        if (b) {
+            if (latin1) {
+                ByteArrayLittleEndian.setInt(val, count, Constants.TRUE_LATIN1);
             } else {
-                val[count++] = 'f';
-                val[count++] = 'a';
-                val[count++] = 'l';
-                val[count++] = 's';
-                val[count++] = 'e';
+                ByteArrayLittleEndian.setLong(val, count << 1, Constants.TRUE_UTF16);
             }
+            count += 4;
         } else {
-            if (b) {
-                count = StringUTF16.putCharsAt(val, count, 't', 'r', 'u', 'e');
+            if (latin1) {
+                ByteArrayLittleEndian.setInt(val, count, Constants.FALS_LATIN1);
+                val[count + 4] = 'e';
             } else {
-                count = StringUTF16.putCharsAt(val, count, 'f', 'a', 'l', 's', 'e');
+                ByteArrayLittleEndian.setLong(val, count << 1, Constants.FALS_UTF16);
+                StringUTF16.putChar(val, count + 4, 'e');
             }
```

Comment:
Are the `StringUTF16​::putCharsAt(…)` methods still used anywhere after this?

Reply:
> Are the `StringUTF16​::putCharsAt(…)` methods still used anywhere after this?

It is used in the test code and should be deleted.